lib/Bindings/Python/circt/dialects/_seq_ops_ext.py

```diff
@@ -1,4 +1,53 @@
-from mlir.ir import OpView, StringAttr
+from circt.support import BackedgeBuilder, NamedValueBuilder
+
+from mlir.ir import IntegerType, OpView, StringAttr, Value
+
+
+class CompRegBuilder(NamedValueBuilder):
```

Comment:
This'll be pulled out and be made more generic in the future (as we discussed), yes?

I have a few thoughts on improvements to backedge builder which may simplify this a bit, but don't let that hold you up.

Reply:
Yep the plan is to keep generalizing as we go.